ethereum_history_api/circuits/lib/src/verifiers/transaction_test.nr

```diff
@@ -0,0 +1,52 @@
+mod assert_tx_proof {
+    use crate::verifiers::transaction::assert_tx_proof;
+    use crate::fixtures::mainnet::{
+        cancun::small_block::{header::transactions_root, transaction_proof::proof},
+        homestead::fork::{header::transactions_root as legacy_transactions_root, transaction_proof::proof as legacy_proof}
+    };
+    use crate::misc::arrays::alter_array;
+
+    #[test]
+    fn success() {
+        assert_tx_proof(proof, transactions_root);
+    }
+
+    #[test]
+    fn legacy_tx() {
+        assert_tx_proof(legacy_proof, legacy_transactions_root);
+    }
+
+    #[test(should_fail_with="Internal node hash does not match the hash extracted from the preceding node")]
+    fn invalid_tx_root() {
+        assert_tx_proof(proof, legacy_transactions_root);
+    }
+
+    #[test(should_fail_with="Arrays do not match at specified offset")]
+    fn invalid_value() {
+        let mut proof_invalid_value = proof;
+        proof_invalid_value.value[proof_invalid_value.value.len() - 1] += 1;
+        assert_tx_proof(proof_invalid_value, transactions_root);
+    }
+
+    #[test(should_fail_with="Internal node hash does not match the hash extracted from the preceding node")]
+    fn invalid_proof() {
+        let mut proof_invalid_proof = proof;
+        proof_invalid_proof.proof[0] += 1;
```

Comment:
Let's change proof at last index instead of first, to see that this fails on a deeper level

Reply:
Thanks. Added a separate test to test for a leaf node. Left this one as internal node test